tfhe/src/core_crypto/algorithms/test/noise_distribution/lwe_encryption_noise.rs

```diff
@@ -184,3 +193,68 @@ fn lwe_compact_public_encrypt_noise_distribution_custom_mod<
 create_parametrized_test!(lwe_compact_public_encrypt_noise_distribution_custom_mod {
     TEST_PARAMS_4_BITS_NATIVE_U64
 });
+
+fn random_noise_roundtrip<Scalar: UnsignedTorus + CastInto<usize>>(params: TestParams<Scalar>) {
+    let mut rsc = TestResources::new();
+    let noise = params.glwe_modular_std_dev;
+    let ciphertext_modulus = params.ciphertext_modulus;
+    let encryption_rng = &mut rsc.encryption_random_generator;
+    let expected_variance = Variance(noise.get_variance());
+
+    let num_ouptuts = 100_000;
+
+    let mut output: Vec<_> = vec![Scalar::ZERO; num_ouptuts];
+
+    encryption_rng.fill_slice_with_random_noise_custom_mod(&mut output, noise, ciphertext_modulus);
+
+    assert!(check_clear_content_respects_mod(
+        &output,
+        ciphertext_modulus
+    ));
+
+    for val in output.iter().copied() {
+        if ciphertext_modulus.is_native_modulus() {
+            let float_torus = val.into_torus();
+            let from_torus = Scalar::from_torus(float_torus);
+            assert!(
+                modular_distance(val, from_torus)
+                    < (Scalar::ONE << (Scalar::BITS.saturating_sub(f64::MANTISSA_DIGITS as usize))),
+                "val={val}, from_torus={from_torus}, float_torus={float_torus}"
+            );
+        } else {
+            let custom_modulus_as_scalar: Scalar =
+                ciphertext_modulus.get_custom_modulus().cast_into();
```

Comment:
it's the conversion function for the UnsignedInteger trait, it basically takes the represented integer and casts it to the destination type if it is compatible, you cant do Scalar as u64 because it's not a basic type of rust